src/hooks/useCurrencySwapper.js

```diff
@@ -235,6 +248,81 @@ const useCurrencySwapper = ({
     }
   }
 
+  const _swapCurve = async (swapAmount, minSwapAmount, isGasEstimate) => {
+    const swapParams = [
+      curveXUSDMetaPool.address,
+      curveMetapoolUnderlyingCoins.indexOf(
+        _maybeToAvToken(coinContract.address)
+      ),
+      curveMetapoolUnderlyingCoins.indexOf(
+        _maybeToAvToken(coinToReceiveContract.address)
+      ),
+      swapAmount,
+      minSwapAmount,
+    ]
+
+    const gasLimit = increaseGasLimitByBuffer(
+      await curveZapper.estimateGas.exchange_underlying(...swapParams, {
+        from: account,
+      }),
+      curveGasLimitBuffer
+    )
+
+    if (isGasEstimate) {
+      return gasLimit
+    } else {
+      return await curveZapper.exchange_underlying(...swapParams, {
+        gasLimit,
+      })
+    }
+  }
+
+  const swapCurveGasEstimate = async (swapAmount, minSwapAmount) => {
+    return (await _swapCurve(swapAmount, minSwapAmount, true)).toNumber()
+  }
+
+  const swapCurve = async () => {
+    const { minSwapAmount: minSwapAmountReceived } = calculateSwapAmounts(
+      outputAmount,
+      coinToReceiveDecimals,
+      priceToleranceValue
+    )
+
+    return {
+      result: await _swapCurve(swapAmount, minSwapAmountReceived, false),
+      swapAmount,
+      minSwapAmount,
+    }
+  }
+
+  const _maybeToAvToken = (address) => {
+    // TODO: do this not in a stupid way, perhaps using coins and underlying coins call?
```

Comment:
Can you think of a better way to do this here? I do not like this.

1) put addresses in contractAddresses.js and pull in by name
2) zip curveXUSDMetaPoolUnderlyingCoins to the contracts using the known indexes
3) ?

Reply:
I think we should create a single coin info struct and put this translation in there as e.g. `curveZapperAvTokenAddress`

Reply:
created issue